.github/workflows/tests-e2e-upgrade.yaml

```diff
@@ -345,6 +345,7 @@ jobs:
           mkdir -p cilium-junits
 
       - name: Install node local DNS
+        timeout-minutes: 2
```

Comment:
Did you consider whether to add this into the `kubectl rollout status` command below as well / instead? It's usually a bit more friendly to developers who hit a timeout to see a richer error, so if kubectl reported a timeout was reached it might be easier to interpret than if GitHub terminates the execution of the step.